scripts/prepare-packages.js

```diff
@@ -0,0 +1,59 @@
+import fs from 'node:fs'
```

Comment:
what do we need this file for?

Reply:
Added in the commit description. 

It renames the output `.d.ts` files in the `cjs` folder to `.d.cts` for better compatibility.

Reply:
If the files are the same as .d.ts and only renamed, what difference can it make what extension they have?

Reply:
`cts` and `mts` files are a forced way to infer the module type.  Since the library's root is already `esm` , the .d.ts will be considered ESM first. To make sure it doesn't do that for the type files in `cjs`, the renaming is being done. 

More about it here:
https://www.typescriptlang.org/docs/handbook/modules/reference.html#module-format-detection